engine/src/main/java/org/terasology/utilities/procedural/SimplexNoise.java

```diff
@@ -62,42 +62,57 @@
     private static final float F4 = ((float) Math.sqrt(5.0f) - 1.0f) / 4.0f;
     private static final float G4 = (5.0f - (float) Math.sqrt(5.0f)) / 20.0f;
 
-    private final short[] perm = new short[512];
-    private final short[] permMod12 = new short[512];
+    private final short[] perm;
+    private final short[] permMod12;
+    private final int permCount;
 
     /**
-     * Initialize permutations with a given seed
+     * Multiply this with the gridDim provided and noise(x,x) will give tileable 1D noise which will tile
+     * when x crosses a multiple of (this * gridDim)
+     */
+    public final float tileable1DMagicNumber = 0.5773502691896258f;
+
+    /**
+     * Initialize permutations with a given seed and grid dimension.
+     * Supports 1D tileable noise
+     * @see SimplexNoise#tileable1DMagicNumber
      *
      * @param seed a seed value used for permutation shuffling
+     * @param gridDim gridDim x gridDim will be the number of squares in the square grid formed after skewing the simplices belonging to once "tile"
      */
     public SimplexNoise(long seed) {
+        this(seed, 8);
```

Comment:
Why does Perlin noise have a grid dimension of `1 << 8` and Simplex noise a grid dimension of `8`?

Reply:
Oh this was a bad miss, fixing this rn thanks!